packages/backend/src/open_payments/invoice/service.ts

```diff
@@ -4,10 +4,13 @@ import { BaseService } from '../../shared/baseService'
 import { Pagination } from '../../shared/pagination'
 import { EventType, WebhookService } from '../../webhook/service'
 import assert from 'assert'
-import { Transaction } from 'knex'
+import { QueryBuilder, Transaction } from 'knex'
 import { ForeignKeyViolationError, TransactionOrKnex } from 'objection'
+import { v4 as uuid } from 'uuid'
 
 export const POSITIVE_SLIPPAGE = BigInt(1)
+// First retry waits 10 seconds, second retry waits 20 (more) seconds, etc.
```

Comment:
And note that 30 follows 20 (i.e. the increase on retry is linear, not geometric).